lib/active_record/connection_adapters/postgis/quoting.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module ActiveRecord
+  module ConnectionAdapters
+    module PostGIS
+      module Quoting
+        def type_cast(value)
+          case value
+          when RGeo::Feature::Instance
+            value.to_s
```

Comment:
After writing https://github.com/rgeo/activerecord-postgis-adapter/issues/402#issuecomment-3071348975 I think perhaps this needs to be extended to support `RGeo::Geos` geometries.  I am unable to do this now but documenting here in case it helps someone in the future.